mathics/builtin/box/inout.py

```diff
@@ -465,41 +477,38 @@ def apply_list(self, boxes, evaluation):
         return result
 
     def init(self, *items, **kwargs):
-        # TODO: check that each element is an string or a BoxConstruct
+        # TODO: check that each element is an string or a BoxExpression
         self.box_options = {}
         if isinstance(items[0], Expression):
             if len(items) != 1:
                 raise Exception(
-                    items, "is not a List[] or a list of Strings or BoxConstruct"
+                    items, "is not a List[] or a list of Strings or BoxExpression"
                 )
             if items[0].has_form("List", None):
                 items = items[0]._elements
             else:
                 raise Exception(
-                    items, "is not a List[] or a list of Strings or BoxConstruct"
+                    items, "is not a List[] or a list of Strings or BoxExpression"
                 )
         for item in items:
-            if not isinstance(item, BoxConstruct):
+            if not isinstance(item, BoxExpression):
                 raise Exception(
-                    item, "is not a List[] or a list of Strings or BoxConstruct"
+                    item, "is not a List[] or a list of Strings or BoxExpression"
                 )
         self.items = tuple(items)
 
-        elements = []
-        for item in items:
-            if isinstance(item, BoxConstruct):
-                elements.append(item.to_expression())
-            else:
-                elements.append(item)
-        elements = Expression(SymbolList, *elements)
-        self._elements = tuple((elements,))
+        self._elements = None
 
     def to_expression(self):
         """
         returns an evaluable expression.
         """
-        items = (item.to_expression() for item in self.items)
-        result = Expression("RowBox", Expression("List", *items))
+        if self._elements is None:
+            self._elements = tuple(
+                item.to_expression() if isinstance(item, BoxExpression) else item
+                for item in self.items
+            )
+        result = Expression("RowBox", Expression("List", *self._elements))
```

Comment:
Will be converting `Expresssion("List",...)` and `Expression(SymboList,...)` kinds of things soon, so there isn't a need for tthis to get changed.

I'll probably also throw in `SymbolRowBox` as well then too.